share/p2p/discovery/manager.go

```diff
@@ -0,0 +1,51 @@
+package discovery
+
+import (
+	"context"
+	"errors"
+)
+
+type Manager struct {
+	discs []*Discovery
+
+	cancel context.CancelFunc
+}
+
+func NewManager(discs []*Discovery) *Manager {
+	return &Manager{
+		discs: discs,
+	}
+}
+
+func (m *Manager) Start(context.Context) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	m.cancel = cancel
+
+	for _, d := range m.discs {
+		err := d.Start(ctx)
```

Comment:
Please, use context from Start function call instead of the Advertisement context(WithCancel).

Reply:
context from start function is short lived

Reply:
yes, and the Start context is meant to be passed to all Start calls

Reply:
Okay I will, but that context is ignored inside discovery.Start anyway in favour of an internal context. It just spawns a bunch of routines.

Reply:
discovery.Start call does not use context indeed, but it expects short lived one. What is the reason to use incorrect context here?